### PR TITLE
Prepare v0.52.0 release

### PR DESCRIPTION
**Pull request by ashutosh-narkar**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,130 @@
...
+
+
+### Website + Ecosystem
+
+- Ecosystem:
+  - Reposaur ([#5854](https://github.com/open-policy-agent/opa/pull/5854)) authored by @charlieegan3
+
+- Website:
+  - Reorganize the `MISCELLANEOUS` section to improve content navigation ([#4614](https://github.com/open-policy-agent/opa/issues/4614)) authored by @lakhanjindam
+  - Update logo for Torque integration ([#5810](https://github.com/open-policy-agent/opa/pull/5810)) authored by @shirabendor-quali
```
Comment (srenatus) [inline CHANGELOG.md:119]: Isn't that ecosystem, too?

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,130 @@
...
+
+### Miscellaneous
+
+- Dependency bumps, notably:
+  - golang from 1.20.2 to 1.20.3
+  - aquasecurity/trivy-action from 0.9.2 to 0.10.0
+  - github.com/spf13/cobra from 1.6.1 to 1.7.0
+  - golang.org/x/net from 0.8.0 to 0.9.0
+  - github.com/prometheus/client_golang from 1.14.0 to 1.15.0
```
Comment (srenatus) [inline CHANGELOG.md:128]: I think only Golang, X/net and prometheus_client are notable here 🤔

Comment (ashutosh-narkar) [reply CHANGELOG.md:126]: Updated

File: CHANGELOG.md (hunk 1 above)
Comment (ashutosh-narkar) [reply CHANGELOG.md]: Moved to ecosystem

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,7 +3,128 @@
...
+
+### Topdown and Rego
+
+- ast: Disallow partial object rules to have other partial object rule within their immediate extent ([#5855](https://github.com/open-policy-agent/opa/issues/5855)) authored by @johanfylling
+- ast: Disallow multi-value rules to have other rules in their extent ([#5813](https://github.com/open-policy-agent/opa/issues/5813)) authored by @johanfylling
+- ast: Set result of groundness check on indexer's AllRules func so that rule evaluation for complete rules is not skipped ([#5857](https://github.com/open-policy-agent/opa/issues/5857)) authored by @ashutosh-narkar
+- rego: Fix duplicate text in error message during module parsing ([#5837](https://github.com/open-policy-agent/opa/pull/5837)) authored by @TzlilSwimmer123
+- planner: Fix bug in call_dynamic with overlapping ref rules ([#5839](https://github.com/open-policy-agent/opa/pull/5839))  authored by @srenatus
+- planner: Fix p.curr <-> prev handling in CallDynamic optimization case ([#5829](https://github.com/open-policy-agent/opa/pull/5829))  authored by @srenatus
```
Comment (srenatus) [inline CHANGELOG.md:102]: I'd combine these into "planner bugs" that have an impact on IR and Wasm usage. As it stands, it's probably gibberish to most users

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,7 +3,128 @@
...
+
+
+check_signature {
+    secret_key := opa.runtime().env.GITHUB_SECRET_KEY
+    hash_body := crypto.hmac.sha256(http_request.raw_body, secret_key)
+    expected_signature := concat("", ["sha256=", hash_body])
+    header_signature = http_request.headers["X-Hub-Signature-256"]
+    crypto.hmac.equal(header_signature, expected_signature)
+}
+```
```
Comment (srenatus) [inline CHANGELOG.md:47]: That's an interesting example -- it clearly has some real world use case behind it; github workflow webhooks. Worth mentioning, maybe?

**Review by srenatus — DISMISSED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (ashutosh-narkar) [reply CHANGELOG.md:102]: Can you suggest some text here?

Comment (ashutosh-narkar) [reply CHANGELOG.md]: Combined the planner bugs. Let me know if the text looks fine.

File: CHANGELOG.md (hunk 4 above)
Comment (ashutosh-narkar) [reply CHANGELOG.md:54]: Added details about a real world use-case.

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,7 +3,134 @@
...
+
+Below is a real world example of how this built-in function can be utilized. Imagine our server is registered as a
+GitHub webhook which subscribes to certain events on GitHub.com. Now we want to limit requests to those coming from GitHub.
+One of the ways to do that is to first set up a secret token and validate the information. Once we create the token on GitHub,
+we'll set up an environment variable that stores this token and make it available to OPA via the `opa.runtime` built-in.
+In the case of GitHub webhooks the validation is done by comparing the hash signature received in the `X-Hub-Signature-256`
+header and calculating a hash using the secret token and payload body. The `check_signature` rule implements this logic.
+Using a plain `==` operator is not advised due to security concerns and hence the new `crypto.hmac.equal` built-in function
+leverages [Go's hmac.Equal](https://pkg.go.dev/crypto/hmac#Equal) which compares two MACs for equality without
+leaking timing information. This helps mitigate certain timing attacks against regular equality operators.
```
Comment (srenatus) [inline CHANGELOG.md:34]: I'm not feeling strongly about this, but I'd perhaps drop these sentences. Proofing Rego code against timing attacks, or even suggesting that if you use this hmac.equal, you'll be safe of such attack vectors doesn't seem like something we should stress 🤔

File: CHANGELOG.md (hunk 3 above)
Comment (srenatus) [reply CHANGELOG.md]: Sorry I hadn't seen your comment earlier. Your text is good, thank you!

File: CHANGELOG.md (hunk 5 above)
Comment (ashutosh-narkar) [reply CHANGELOG.md]: Removed those sentences.

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,7 +3,131 @@
...
-## Unreleased
+## 0.52.0
+
+This release contains some enhancements, bugfixes, and a new builtin function.
+
+### Allow Adding Labels via Discovery
+
+Previously OPA did not allow any updates to the labels provided in the boot configuration via the discovered (ie. service)
+config. This was done to avoid breaking the discovery configuration. But there are use cases where labels can serve as a convenient
+way to pass information that could be used for example in policies, status updates or decision logs. This change allows
```
Comment (philipaconrad) [inline CHANGELOG.md:14]:
```suggestion
way to pass information that could be used in policies, status updates or decision logs. This change allows
```
[suggestion]

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,7 +3,131 @@
...
+additional labels to be configured in the service config which are then made available during runtime.
+
+See [the Discovery documentation](https://www.openpolicyagent.org/docs/v0.52.0/management-discovery/#limitations)
+for more details.
+
+Authored by @mjungsbluth.
+
+### New Built-In Function: crypto.hmac.equal
+
+`crypto.hmac.equal` provides a convenient way to compare hashes generated by MD5, SHA-1, SHA-256 and SHA-512 hashing algorithms.
```
Comment (philipaconrad) [inline CHANGELOG.md:24]:
```suggestion
`crypto.hmac.equal` provides a convenient way to compare hashes generated by the MD5, SHA-1, SHA-256 and SHA-512 hashing algorithms.
```
[suggestion/nit]

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,7 +3,131 @@
...
+
+Below is a real world example of how this built-in function can be utilized. Imagine our server is registered as a
+GitHub webhook which subscribes to certain events on GitHub.com. Now we want to limit requests to those coming from GitHub.
+One of the ways to do that is to first set up a secret token and validate the information. Once we create the token on GitHub,
+we'll set up an environment variable that stores this token and make it available to OPA via the `opa.runtime` built-in.
```
Comment (philipaconrad) [inline CHANGELOG.md:29]:
```suggestion
we'll set up an environment variable that stores this token and makes it available to OPA via the `opa.runtime` built-in.
```
[minor copyedit]

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,7 +3,131 @@
...
+    secret_key := opa.runtime().env.GITHUB_SECRET_KEY
+    hash_body := crypto.hmac.sha256(http_request.raw_body, secret_key)
+    expected_signature := concat("", ["sha256=", hash_body])
+    header_signature = http_request.headers["X-Hub-Signature-256"]
+    crypto.hmac.equal(header_signature, expected_signature)
+}
+```
+
+See [the documentation on the new built-in](https://www.openpolicyagent.org/docs/v0.52.0/policy-reference/#builtin-crypto-cryptohmacequal)
+for all details.
```
Comment (philipaconrad) [inline CHANGELOG.md:54]:
```suggestion
for all the details.
```
[copyedit]

**Review by philipaconrad — APPROVED** [review]
This looks great, @ashutosh-narkar! Only minor copy editing suggestions below:

File: CHANGELOG.md (hunk 9 above)
Comment (ashutosh-narkar) [reply CHANGELOG.md]: Done

File: CHANGELOG.md (hunk 8 above)
Comment (ashutosh-narkar) [reply CHANGELOG.md]: Done

File: CHANGELOG.md (hunk 7 above)
Comment (ashutosh-narkar) [reply CHANGELOG.md]: Done

File: CHANGELOG.md (hunk 6 above)
Comment (ashutosh-narkar) [reply CHANGELOG.md]: Done.

**Pull request merged**